src/sagas.js

```diff
@@ -282,27 +320,66 @@ function* fetchVmsByCountVLower (action) {
       yield fetchVmsCDRom({ vms: internalVms })
       yield fetchVmsNics({ vms: internalVms })
     } else {
-      logDebug('fetchVmsByCount() shallow fetch requested - skipping other resources')
+      logDebug('fetchVmsByCountVLower() shallow fetch requested - skipping other resources')
     }
   }
 
   yield put(persistState())
+  return fetchedVmIds
+}
+
+export function* fetchSingleVm (action) {
+  const { vmId, shallowFetch } = action.payload
+  yield startProgress({ vmId, name: 'refresh_single' })
+
+  const isOvirtGTE42 = compareVersionToCurrent({ major: 4, minor: 2 })
+  if (isOvirtGTE42 && !shallowFetch) {
+    action.payload.additional =
+      ['cdroms', 'sessions', 'disk_attachments.disk', 'graphics_consoles', 'nics']
+  }
+
+  yield put(addVmsFetchedById({ vmIds: [ vmId ] }))
+  const vm = yield callExternalAction('getVm', Api.getVm, action, true)
+  let internalVm = false
```

Comment:
Can we initialize it to `null`/`undefined` or not initialize it at all? I believe it's beneficial to keep the variable type as narrow as possible.